.github/workflows/test.yml

```diff
@@ -8,44 +8,38 @@ jobs:
       runs-on: ubuntu-latest
       strategy:
         matrix:
-          python: ['3.7', '3.8', '3.9', '3.10', '3.11']
+          python: ['3.8', '3.9', '3.10', '3.11', '3.12']
```

Comment:
We may still need 3.7 here for DLRN.

Reply:
OK, I'll wait to see what @evallesp says on that. 3.7 has been EOL for almost a year [1], and we have removed it from the rest of our projects. I thought the jobs started failing due to 3.7 not being available on the ci-runner, but I can add it back if needed and see what happens (I could be remembering wrong).

[1] https://devguide.python.org/versions/

Reply:
That's right, the 3.7 jobs failed in the other open PR and we had to remove them.

Reply:
I am fine removing it, just wanted to make sure we are not losing coverage we may need.

Reply:
Right! We can't use 3.7 for jobs with Github Actions at this time so I don't think anything is lost.

@eggmaster @evallesp @fuzzball81 This is the last of these clean-up patches, if any of you has the time to look it over for a second approval!